Common/Securities/Future/FuturesExpiryUtilityFunctions.cs

```diff
@@ -96,6 +100,68 @@ public static DateTime NthLastBusinessDay(DateTime time, int n)
             return lastDayOfMonth.AddDays(-totalDays);
         }
 
+        /// <summary>
+        /// Calculates the n^th business day of the month (includes checking for holidays)
+        /// </summary>
+        /// <param name="time">Month to calculate business day for</param>
+        /// <param name="nthBusinessDay">n^th business day to get</param>
+        /// <param name="additionalHolidays">Additional user provided holidays to not count as business days</param>
+        /// <returns>Nth business day of the month</returns>
+        public static DateTime NthBusinessDay(DateTime time, int nthBusinessDay, IEnumerable<DateTime> additionalHolidays = null)
```

Comment:
This may be better suited living in `Time.cs` as it's not specific to futures. Also, the `additionalHolidays` would be better represented as an `ICollection<DateTime>` or even better as a `HashSet<DateTime>` -- `IEnumerable.Contains` is an O(n) operation,  so performing it in a loop moves to n^m where m is the number of loops, whereas `HashSet.Contains` is O(1) -- _also_ since the parameter is of type `IEnumerable<DateTime>` its possible for a caller to provide a lazy enumerable which will be re-evaluated on each call to `Contains` (hence the `ICollection<DateTime>` suggestion) -- my preference would be `HashSet<DateTime>` which _should_ be the native form of holding the collection of holiday dates.

Reply:
After further review, I understand why this was placed here give the `NotHoliday` method also exists here -- since both of these methods are tightly bound to `USHolidays`, might want to make that distinction via `NthUsBusinessDay` or modify the method to accept a complete list of holiday so it's agnostic to the market.